### DISCUSSION TITLE
Where to find the documentation about error responses from token refresh API

### DISCUSSION THREAD

**Discussion by PauloKoglin** [keycloak/keycloak#27007] [Q&A]
Hi 👋

Is there any documentation, which lists the possible http error responses when calling an endpoint like `openid-connect/token`?

I'm trying to handle token errors, but I can't find the documentation, which explains why (use case) the error occurs.

Example:
When `/realms/<my-client>/protocol/openid-connect/token` gets called with some invalid state, I get errors like these, but I can't figure it out why:

```
{
  "headers":{
    "normalizedNames":{ },
    "lazyUpdate":null
  },
  "status":400,
  "statusText":"Bad Request",
  "url":"http://localhost:8080/realms/my-project/protocol/openid-connect/token",
  "ok":false,
  "name":"HttpErrorResponse",
  "message":"Http failure response for http://localhost:8080/realms/my-project/protocol/openid-connect/token: 400 Bad Request",
  "error":{
    "error":"invalid_grant",
    "error_description":"Token is not active"
  }
}
```

or

```
{
  "headers":{
    "normalizedNames":{ },
    "lazyUpdate":null
  },
  "status":400,
  "statusText":"Bad Request",
  "url":"http://localhost:8080/realms/my-project/protocol/openid-connect/token",
  "ok":false,
  "name":"HttpErrorResponse",
  "message":"Http failure response for http://localhost:8080/realms/my-project/protocol/openid-connect/token: 400 Bad Request",
  "error":{
    "error":"invalid_grant",
    "error_description":"Stale token"
  }
}
```

or just

```
"Token has expired"
```

I wish there was a table of error responses from endpoints :) 
Any idea where to find it, if it exists?

**Comment by shawkins** [comment]
Relates to https://github.com/keycloak/keycloak/issues/37961 - some additional logging or docs could help self-service token problems.
